sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample11_MockEventHubProducerClientTests.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Diagnostics;
+using Azure.Messaging.EventHubs.Producer;
+using Moq;
+using NUnit.Framework;
+
+namespace Azure.Messaging.EventHubs.Tests.Snippets
+{
+    /// <summary>
+    ///   The suite of tests defining the snippets used in the Event Hubs
+    ///   Sample10_AzureEventSourceListener sample.
+    /// </summary>
+    ///
+    [TestFixture]
+    [Category(TestCategory.Live)]
+    [Category(TestCategory.DisallowVisualStudioLiveUnitTesting)]
+    public class Sample11_MockEventHubProducerClientTests
+    {
+        /// <summary>
+        ///   Performs basic smoke test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public void EventDataBatch()
+        {
+            #region Snippet:EventHubs_Sample11_EventDataBatch
+
+            // Data batch mock simulating failure adding events to it.
+
+            var mockEventDataBatch = EventHubsModelFactory.EventDataBatch(
```

Comment:
nice....  mocking a batch gets asked about a lot!   However, I'm pretty sure this won't compile.   It would need to be something like:

```csharp

var eventsInBatch = new List<EventData>();
var mockEventDataBatch = EventHubsModelFactory.EventDataBatch(
    500,
    eventsInBatch,
    null,
    _ => false);
```